app/components/huddle-reason-icon.js

```diff
@@ -0,0 +1,57 @@
+import Component from 'ember-component';
+import computed from 'ember-computed';
+import { REASON_CODES } from 'ember-on-fhir/models/huddle-patient';
+
+export default Component.extend({
+  tagName: 'i',
+  classNameBindings: ['reasonIconClass'],
+
+  huddle: null,
+  patient: null,
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    this.$().tooltip({
+      container: 'body',
+      title: () => {
+        return this.get('huddlePatient.reasonText');
+      }
+    });
+  },
+
+  willDestoryElement() {
+    this._super(...arguments);
+    this.$().tooltip('destroy');
+  },
+
+  huddlePatient: computed('huddle', 'patient', {
+    get() {
+      let huddle = this.get('huddle');
+      if (huddle) {
+        return huddle.getHuddlePatient(this.get('patient'));
+      }
+    }
+  }),
+
+  reasonIconClass: computed('huddlePatient.reason', {
+    get() {
+      let reason = this.get('huddlePatient.reason');
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch would be a much cleaner implementation of this